pkg/csource/options.go

```diff
@@ -204,63 +204,55 @@ func (opts Options) Serialize() []byte {
 	return data
 }
 
-func deserializeLegacyOptions1(data string, opts *Options) error {
-	waitRepeat, debug := false, false
-	n, err := fmt.Sscanf(data,
-		"{Threaded:%t Collide:%t Repeat:%t Procs:%d Sandbox:%s"+
-			" Fault:%t FaultCall:%d FaultNth:%d EnableTun:%t UseTmpDir:%t"+
-			" HandleSegv:%t WaitRepeat:%t Debug:%t Repro:%t}",
-		&opts.Threaded, &opts.Collide, &opts.Repeat, &opts.Procs, &opts.Sandbox,
-		&opts.Fault, &opts.FaultCall, &opts.FaultNth, &opts.NetInjection, &opts.UseTmpDir,
-		&opts.HandleSegv, &waitRepeat, &debug, &opts.Repro)
-
-	if err == nil {
-		if want := 14; n != want {
-			return fmt.Errorf("failed to parse repro options: got %v fields, want %v", n, want)
-		}
+func deserializeLegacyOptions(data string, opts *Options) (int, error) {
+	ignoreBool := true
+	keyToTarget := map[string]struct {
+		typeFormat  string
+		destination any
+	}{
+		"Threaded":      {"%t", &opts.Threaded},
+		"Collide":       {"%t", &opts.Collide},
+		"Repeat":        {"%t", &opts.Repeat},
+		"Procs":         {"%d", &opts.Procs},
+		"Sandbox":       {"%s", &opts.Sandbox},
+		"SandboxArg":    {"%d", &opts.SandboxArg},
+		"Fault":         {"%t", &opts.Fault},
+		"FaultCall":     {"%d", &opts.FaultCall},
+		"FaultNth":      {"%d", &opts.FaultNth},
+		"EnableTun":     {"%t", &opts.NetInjection},
+		"UseTmpDir":     {"%t", &opts.UseTmpDir},
+		"EnableCgroups": {"%t", &opts.Cgroups},
+		"HandleSegv":    {"%t", &opts.HandleSegv},
+		"WaitRepeat":    {"%t", &ignoreBool},
+		"Debug":         {"%t", &ignoreBool},
+		"Repro":         {"%t", &opts.Repro},
 	}
-	return err
-}
 
-func deserializeLegacyOptions2(data string, opts *Options) error {
-	waitRepeat, debug := false, false
-	n, err := fmt.Sscanf(data,
-		"{Threaded:%t Collide:%t Repeat:%t Procs:%d Sandbox:%s"+
-			" Fault:%t FaultCall:%d FaultNth:%d EnableTun:%t UseTmpDir:%t"+
-			" EnableCgroups:%t HandleSegv:%t WaitRepeat:%t Debug:%t Repro:%t}",
-		&opts.Threaded, &opts.Collide, &opts.Repeat, &opts.Procs, &opts.Sandbox,
-		&opts.Fault, &opts.FaultCall, &opts.FaultNth, &opts.NetInjection, &opts.UseTmpDir,
-		&opts.Cgroups, &opts.HandleSegv, &waitRepeat, &debug, &opts.Repro)
-	if err == nil {
-		if want := 15; n != want {
-			return fmt.Errorf("failed to parse repro options 2: got %v fields, want %v", n, want)
+	data = strings.ReplaceAll(data, "{", "")
+	data = strings.ReplaceAll(data, "}", "")
+	tokens := strings.Split(data, " ")
+	totalRead := 0
+	for _, token := range tokens {
+		keyValuePair := strings.Split(token, ":")
```

Comment:
Nit: `key, value, found := strings.Cut(token, ":")`

Reply:
Thanks for sharing. It removes 2 lines.
Like.